frontends/web/src/routes/account/send/services.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { alertUser } from '../../../components/alert/Alert';
+import { i18n } from '../../../i18n/i18n';
+
+export type TProposalError = {
+    addressError: string;
+    amountError: string;
+    feeError: string;
+}
+
+export const txProposalErrorHandling = (registerEvents: () => void, unregisterEvents: () => void, errorCode?: string) => {
+  const { t } = i18n;
+  switch (errorCode) {
+  case 'invalidAddress':
+    return { addressError: t('send.error.invalidAddress') };
+  case 'invalidAmount':
+  case 'insufficientFunds':
+    return { amountError: t(`send.error.${errorCode}`), proposedFee: undefined };
+  case 'feeTooLow':
+  case 'feesNotAvailable':
+    return { feeError: t(`send.error.${errorCode}`) };
+  default:
+    if (errorCode) {
+      unregisterEvents();
+      alertUser(errorCode, { callback: registerEvents });
+    }
```

Comment:
add a `return { proposedFee: undefined };` after the condition, then it looks to me that it should behave the same as before this refactoring.

Reply:
Oh good catch, ty!

Reply:
actually, looking at it again, returning the proposedFee after the if condition seems to be a bit of a different flow than the original one.

Original: first set the state, then if statement.

Refactor: if statement then set the state.

To me the difference looks harmless. Wanna make sure by testing it but unsure how one can reach the `default` block (with `errorCode`). WDYT? Any concerns?

Reply:
true, the order is a bit different, but it looks good to me.